### PR TITLE
[WIP] Lower long compares that are materialized into a register for ARM32

### PR DESCRIPTION
**Pull request by mikedn**
Port #10275 to ARM32.

Any feedback appreciated, I cannot actually test this so I can only assume that it's correct based on JitDisasm output and my limited knowledge of ARM32 assembly.

### CODE REVIEW

**Comment by mikedn** [comment]
Sample disasm: https://gist.github.com/mikedn/9337a69a90f1ad1eec52f003e5a77ac8

**Comment by hqueue** [comment]
@mikedn I've ran `JIT/CodeGenBringUpTests` with `COMPlus_AltJit=* COMPlus_AltJitAssertOnNYI=0` and all tests passed. I think this seems to be working and fixed previous problem.

Can you suggest specific tests for long compare ? , because I'm not sure long compare is tested in JIT/CodeGenBringUpTests.

**Comment by mikedn** [comment]
> because I'm not sure long compare is tested in JIT/CodeGenBringUpTests.

Yes, unfortunately bring up tests tend to stick to `int`. There are a few that do indirectly test `long` and `ulong` equality - Div/Mod/UDiv/UModConst.

> Can you suggest specific tests for long compare

Probably the best ones are in JIT\IL_Conformance. Those are similar to the bring up tests but do test 64 bit IL compare instructions (in particular the ones from JIT\IL_Conformance\Old\Conformance_Base).

There's also JIT\Methodical\int64 though none of those tests seem to be focused on compares. "addsub" tests from signed and unsigned subfolders do use a bit more than just `==` and `!=`.

File: src/jit/emitarm.cpp [hunk 1]
```diff
@@ -7787,6 +7787,17 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
             assert(!"Invalid ins for overflow check");
         }
     }
+
+    if (dst->gtSetFlags())
+    {
+        if ((ins != INS_cmp) && (ins != INS_cmn))
```
Comment (mikedn) [inline src/jit/emitarm.cpp]: This seems bogus, `cmp` and `cmn` are not ternary.

**Comment by mikedn** [comment]
I'll have to add `GT_CMP` to `Lowering::IsContainableImmed`, without it we generate funny code like:
```asm
000134  2300    movs    r3, 0
000136  4298    cmp     r0, r3
```

**Comment by hqueue** [comment]
I've ran `JIT/IL_Conformance/` and found that this PR seems to work well with the test. I found no critical issue in those tests.

**Comment by mikedn** [comment]
Thanks for testing!

I'll wait for #11876 to update `Lowering::IsContainableImmed` before finishing this.

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib

**Comment by pgavlin** [comment]
This will probably need to be adjusted slightly to accommodate the merge of #11872. LGTM otherwise.

File: src/jit/emitarm.cpp (hunk 1 above)
Comment (pgavlin) [inline src/jit/emitarm.cpp]: Yeah, this is odd. @briansull @hqueue any ideas?

Comment (mikedn) [inline src/jit/emitarm.cpp]: That was a note to self to remove the unnecessary `if`. I don't think there's any doubt that `cmp` is not ternary :)

Comment (pgavlin) [inline src/jit/emitarm.cpp]: ...that makes more sense. Thanks! :)

**Comment by mikedn** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

One of the chaos tests timed out again: 
```
MESSAGE:
\ncmdLine:D:\\j\\workspace\\debug_windows---17180f1b\\bin\\tests\\Windows_NT.x64.Debug\\JIT\\Generics\\Coverage\\chaos65204782cs_o\\chaos65204782cs_o.cmd Timed Out\r\n\n\nReturn code: -100\nRaw output file: D:\\j\\workspace\\debug_windows---17180f1b\\bin\\tests\\Windows_NT.x64.Debug\\Reports\\JIT.Generics\\Coverage\\chaos65204782cs_o\\chaos65204782cs_o.output.txt\nRaw output:\nBEGIN EXECUTION\r\n \"D:\\j\\workspace\\debug_windows---17180f1b\\tests\\..\\bin\\tests\\Windows_NT.x64.Debug\\Tests\\Core_Root\\corerun.exe\" chaos65204782cs_o.exe \r\nChaos.A0\r\nChaos.A0\r\nChaos.A0\r\nChaos.A0\r\nChaos.A0A0...
```

**Pull request merged**